### PR TITLE
refactor(Core/Config):  Change PlayerStart.AllSpells to .CustomSpells

### PR DESCRIPTION
**Pull request by 55Honey**
## Changes Proposed:
Remove wrong info regarding the playercreateinfo_spell_custom table. Change the config flags name to match its pupose.

## Issues Addressed:
Clarification. 

## Tests Performed:
Builds fine on Win10 64bit

## How to Test the Changes:
- Build the core as usual
- Use `PlayerStart.CustomSpells = 1` in worldserver.conf instead of `PlayerStart.AllSpells = 1`
- Creating new characters will result in starter spells being taken from playercreateinfo_spell_custom table (empty in this PR)

## Known Issues and TODO List:
- [x] gather opinions, if 
1) setting the config flag `PlayerStart.CustomSpells = 1` should **add** the contents of `playercreateinfo_spell_custom` table **on top** of playercreateinfo_spell or
2) if the contents of `playercreateinfo_spell_custom`  should keep **replacing** `playercreateinfo_spell`

## Target Branch(es):
- [x] Master

 
## How to Test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here in the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

### CODE REVIEW

File: src/server/game/World/World.cpp [hunk 1]
```diff
@@ -1279,9 +1279,7 @@ void World::LoadConfigSettings(bool reload)
     sLog->outString("WORLD: VMap support included. LineOfSight:%i, getHeight:%i, indoorCheck:%i PetLOS:%i", enableLOS, enableHeight, enableIndoor, enablePetLOS);
 
     m_bool_configs[CONFIG_PET_LOS]          = sConfigMgr->GetBoolDefault("vmap.petLOS", true);
-    m_bool_configs[CONFIG_START_ALL_SPELLS] = sConfigMgr->GetBoolDefault("PlayerStart.AllSpells", false);
-    if (m_bool_configs[CONFIG_START_ALL_SPELLS])
-        sLog->outString("WORLD: WARNING: PlayerStart.AllSpells enabled - may not function as intended!");
+    m_bool_configs[CONFIG_START_ALL_SPELLS]  = sConfigMgr->GetBoolDefault("PlayerStart.CustomSpells", false);
     m_int_configs[CONFIG_HONOR_AFTER_DUEL]    = sConfigMgr->GetIntDefault("HonorPointsAfterDuel", 0);
```
Comment (Kitzunu) [inline src/server/game/World/World.cpp:1283]: Still missing the space
```suggestion
    m_bool_configs[CONFIG_START_ALL_SPELLS]   = sConfigMgr->GetBoolDefault("PlayerStart.CustomSpells", false);
    m_int_configs[CONFIG_HONOR_AFTER_DUEL]    = sConfigMgr->GetIntDefault("HonorPointsAfterDuel", 0);
```

Comment (55Honey) [reply src/server/game/World/World.cpp]: I've clicked the "Commit suggestion" already, not sure whats wrong.

File: src/server/worldserver/worldserver.conf.dist [hunk 2]
```diff
@@ -3225,15 +3225,14 @@ ItemDelete.ItemLevel = 80
...
 
 #
-#    PlayerStart.AllSpells
-#        Description: If enabled, players will start with all their class spells (not talents).
-#                     You must populate playercreateinfo_spell_custom table with the spells you
-#                     want, or this will not work! The table has data for all classes / races up
-#                     to TBC expansion.
+#    PlayerStart.CustomSpells
+#        Description: If enabled, players will ignore spells from playercreateinfo_spellstart at character
+#                     creation and will start with spells from the table playercreateinfo_spell_custom instead.
```
[the remark this answers is not in the document]
Comment (55Honey) [reply src/server/worldserver/worldserver.conf.dist:3230]: Yes. AzerothCore rev. 879468506f8e 2021-01-25 20:50:53 +0100 (master branch) (Win64, Debug) (worldserver-daemon)
The wording is correct now. The custom table currently replaces the default one.

[the remark this answers is not in the document]
Comment (Kitzunu) [reply src/server/worldserver/worldserver.conf.dist:3230]: then im happy =)

**Review by Kitzunu — APPROVED** [review]

**Comment by Kitzunu** [comment]
As for the question, I believe that it should be kept working as it should for now.

**Comment by BarbzYHOOL** [comment]
thx for the PR!

so this question is mine, and I'm not sure if we should change it (i guess it can be quite easy to do), but there could be real arguments for or against before deciding :D

**Comment by 55Honey** [comment]
First argument against the change: If you add them on top, you can not remove the default spells anymore.

**Comment by BarbzYHOOL** [comment]
as for the question, it's totally arbitrary to have 2 tables for the same thing, we could have 3, 4 or 5 or 1000
but anyway, it's not like it's urgent to change this, i'll open an issue about it later

**Comment by pklloveyou** [comment]
mark

**Comment by pklloveyou** [comment]
Tested

**Comment by Kitzunu** [comment]
thanks

**Pull request merged**